libcloud/compute/drivers/openstack.py

```diff
@@ -2313,15 +2313,20 @@ def ex_get_floating_ip(self, ip):
         ip_obj, = [x for x in floating_ips if x.ip_address == ip]
         return ip_obj
 
-    def ex_create_floating_ip(self, ip_pool):
+    def ex_create_floating_ip(self, ip_pool=None):
         """
-        Create new floating IP
+        Create new floating IP. The ip_pool attribute is optional only if your
+        infrastructure has only one IP pool available.
 
         :param      ip_pool: name of the floating IP pool
         :type       ip_pool: ``str``
 
         :rtype: :class:`OpenStack_1_1_FloatingIpAddress`
         """
+        if (ip_pool is None) and (len(self.ex_list_floating_ip_pools()) == 1):
+            ip_pool = self.ex_list_floating_ip_pools()[0].name
+        elif (ip_pool is None) and (len(self.ex_list_floating_ip_pools()) > 1):
```

Comment:
I'm not sure about this, if the behavior of open-stack changes, we've bound ourselves to this assumption. I would remove this safety check and instead say
```python
data = {} if ip_pool is None else {'pool': ip_pool}
```
keeping it all on one line.